psq/worker.py

```diff
@@ -42,7 +45,12 @@ def _safe_dequeue(self):
             wait_exponential_multiplier=1000, wait_exponential_max=10000,
             retry_on_exception=lambda e: not isinstance(e, KeyboardInterrupt))
         def inner():
-            return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            try:
+                return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            except errors.GaxError as e:
```

Comment:
@dhermes indeed, it seems like the same issue I'm facing. 

I'm not sure whether this is expected and `psq` should be in charge of handling the exception or, as you mention in that thread, just return an empty set from `pubsub`. The latter will remove the necessity of handling the exception in client libraries.